addons/mrp/models/mrp_production.py

```diff
@@ -1256,6 +1256,13 @@ def _get_origin(self):
             origin = '%s,%s' % (origin, self.name)
         return origin
 
+    def _mark_byproducts_as_produced(self):
+        for move in self.move_byproduct_ids:
+            # sudo needed for portal users
+            if move.sudo()._should_bypass_set_qty_producing():
```

Comment:
The `_should_bypass_set_qty_producing` was made to not override quantities set in workorder or component without bom. In case of `picked` we should set it on all byproduct